src/cloudimagedirectory/filter/filter.py

```diff
@@ -15,15 +19,44 @@ def get_utc_datetime(date_string: str) -> pd.Timestamp:  # type: ignore[no-any-u
 def FilterImageByFilename(word: str) -> Callable:
     """Filter images by filename."""
     print("filter images by filename: " + word)
-    return lambda data: [d for d in data if not d.filename.lower().__contains__(word.lower())]
+
+    filtered_image_by_filename_counter = meter.create_counter(
+        name="filtered_image_by_filename_counter",
+        description="Counts the number of filtered images by filename",
+        unit="1",
+    )
+
+    def _filter_image_by_filename(data: str) -> list:
+        result = []
+        for d in data:
+            if not d.filename.lower().__contains__(word.lower()):
```

Comment:
It is better to use `in` instead of `__contains__`. For example: `if word.lower() not in d.filename.lower()`

Reply:
Yes, I remember our meeting around that topic. As we used these method names in many places of the project, I would prefer to keep it here consistent with the other places and change them in an extra PR - What do you think? 
I opened an issue for that topic https://github.com/redhatcloudx/transformer/issues/733

Reply:
Yeah, in this case, it is just a bad practice, but it is definitely good to raise awareness of this issue :-)